scripts/docker/Dockerfile

```diff
@@ -0,0 +1,119 @@
+FROM ubuntu:20.04
+#FROM intel/oneapi-basekit:devel-ubuntu18.04
+
+ENV SOS_INSTALL=/home/sos
+RUN mkdir /home/sos
+ENV DEBIAN_FRONTEND=noninteractive
+ENV CC=gcc
+ENV CXX=g++
+# For OneAPI Compilers use clang and clang++
+  #ENV CC=clang
+  #ENV CXX=clang++
```

Comment:
Unfortunately, I don't think `clang` will work anymore starting with oneAPI v2022.x.  Have you tried it recently?  I'm curious what version is included with this `devel-ubuntu18.04` image?

I can't find any release notes about it yet online, but `compiler/2022.0.2/linux/bin-llvm/README.txt` in the 2022.1.2 Base Toolkit it says: 

```quote
1. If invoking "clang" or "clang++" directly, please try to use icx/icpx/dpcpp
   instead. Direct use of clang as a driver is not supported. The "clang"
   drivers do not have the same behavior as either icx or upstream LLVM, and
   are not widely tested.

2. If for some reason you must use the OneAPI clang/clang++, please report the
   issue. Until the issue is resolved, you can find the clang/clang++ driver's
   location by running "dpcpp --print-prog-name=clang".
```

Anyway, this is probably ok, especially if it still works - just something to keep in mind...

Reply:
`clang` does still work for now, as long as the Dockerfile builder's `PATH` points to the OneAPI compiler bin area where `clang`, `icx`, etc. are (The builder's `PATH` is updated in the Dockerfile using the `ENV` command). However, one thing to note is that when testing out this Dockerfile with the `icx` compiler, the build failed when building ucx due to a NaN and infinity comparison in one of their headers.

Reply:
Ok - thanks for getting to the bottom of that @kholland-intel

Reply:
I'm not quite sure... if you can think of a good upstream fix, but all means go for it (I can't think of an easy fix at the moment, but it must be possible).

Otherwise, I think a comment in the README is probably fine.  We could also just remove it or comment it out... UCX is not being used in the default build right?

Reply:
Correct, UCX is not being used by default. Maybe I can comment it out and put a note in the README that it does not work with the `icx` compiler.